### PR TITLE
Modal dialog responsive view

### PR DESCRIPTION
**Pull request by mehaknasir**
This Pr is created in reference to the ticket (https://openedx.atlassian.net/browse/TNL-8477) which requires modal dialog to be fullscreen on mobile view and on desktop as modal. 
- react responsive version is updated to 8.2.0
- isFullscreenOnMobile prop is passed in the modal dialog component which will help calculate if the fullscreen design is needed on mobile view. 
- docs are updated to match requirements
- updated test case for responsive component
- updated test cases for pagination

### CODE REVIEW

**Comment by asadazam93** [comment]
@mehaknasir Can you add a screenshot of how it will look?

**Comment by mehaknasir** [comment]
> @mehaknasir Can you add a screenshot of how it will look?

https://deploy-preview-782--paragon-edx.netlify.app/components/modal/modal-dialog/
@asadazam93 you can see a preview of the change live by visiting this link.

**Review by awaisdar001 — APPROVED** [review]
Looks good to me.

**Review by abutterworth — APPROVED** [review]

**Review by adamstankiewicz — APPROVED** [review]

File: src/Modal/ModalDialog.jsx [hunk 1]
```diff
@@ -22,7 +23,10 @@ function ModalDialog({
   closeLabel,
   isFullscreenScroll,
   className,
+  isFullscreenOnMobile,
 }) {
+  const isTabletOrMobile = useMediaQuery({ query: '(max-width: 768px)' });
```
[the remark this answers is not in the document]
Comment (mehaknasir) [reply src/Modal/ModalDialog.jsx:28]: @adamstankiewicz I tried that already, I am not sure why but values are not picking up using this syntax hence I have to hardcode to make the query work.

**Pull request merged**